### ISSUE TITLE
Pretraining of AAE_v3 + which data has been used?

### ISSUE THREAD

**Issue by jat9292** [spoilt333/onco-aae#3]
Hello,
I was wondering, why are you imposing in the pretraining of  the AAE the following condition:
e >= 5 ??
Many times the loss of enc begins lower than this value for epoch < 5 but then grows too quickly for the generator to converge. So this condition makes the algorithm slower, hence i don't understand it's purpose?
Also, why pretraining first part of the model and not directly training the complete model?

Another issue concerns the data which has been used in the original article. 
When I donwload the dataset from DTP website : NCI-60 cell line assay full dose response data (by the way the 2014 version you linked is no more available so I downloaded the 2016 one), there are many different values for different experiences with exact same parameters, how to choose the associated GI percentage? 
More precisely, in the Cancer60DoseResponse csv file, are given many values of GIprcnt for same compound with same concentration (even with same panel and cell types) : did you take an average of those values or pick one randomly? 
Did you consider a specific panel and cell numbers? Or did you mix different panels and cells in the model?
How did you deal with the different concentration units?

Also, I had issues computing some of the MACCS fingerprints for many of the molecules using RDKIT. For instance, NSC534, which is in the Cancer60DoseResponse dataset has not a computable MACCS166 fingerprint according to RDKIT. Approximately 2% of given molecules have not a computable fingerprint using RDKIT. How did you deal with this issue?

Thank you in advance.
Best regards,
Joseph.

**Comment by jat9292** [comment]
Also, two other questions related to the training part:
Why do you use 10000 optimization iterations inside of each epochs?
Why do you stop optimizing according to manifold cost? (the "sess.run(train_manifold" part is commented in your code but why?)

**Comment by spoilt333** [comment]
first of all, pretraining is a common stabilisation trick, if you start training whole network from scratch it is much more probable to get discriminator or generator collapse.
research was done for 1 cell type, different GIprcnt could be averaged if you like or sampled as different example and used as source of natural noise in data.
I'm not sure about this question:

> here are many different values for different experiences with exact same parameters, how to choose the associated GI percentage?

different concentrations units should be converted to the same scale of course
we didn't experienced issues with rdkit conversions.
10000 is enough to show progress and not too much
we are not sure at the moment is manifold cost actually very useful, but you can try both with or without it.
